[
  {
    "discussion_id": "2031568051",
    "pr_number": 7710,
    "pr_file": "components/admission-webhook/manifests/base/deployment.yaml",
    "created_at": "2025-04-07T16:04:48+00:00",
    "commented_code": "ports:\n        - name: https-webhook\n          containerPort: 4443\n        securityContext:\n          runAsNonRoot: true\n          allowPrivilegeEscalation: false\n          runAsUser: 1000",
    "repo_full_name": "kubeflow/kubeflow",
    "discussion_comments": [
      {
        "comment_id": "2031568051",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 7710,
        "pr_file": "components/admission-webhook/manifests/base/deployment.yaml",
        "discussion_id": "2031568051",
        "commented_code": "@@ -18,6 +18,15 @@ spec:\n         ports:\n         - name: https-webhook\n           containerPort: 4443\n+        securityContext:\n+          runAsNonRoot: true\n+          allowPrivilegeEscalation: false\n+          runAsUser: 1000",
        "comment_created_at": "2025-04-07T16:04:48+00:00",
        "comment_author": "thesuperzapper",
        "comment_body": "Is there any particular reason that `UID: 1000` was chosen, I think its not correct?\r\n\r\nThe Docker image should run as `65532` in most cases:\r\n- https://github.com/kubeflow/kubeflow/blob/v1.10.0/components/admission-webhook/Dockerfile#L22\r\n\r\nAlso, perhaps we should not specify a `runAsUser` in the first place, as we already have `runAsNonRoot`.",
        "pr_file_module": null
      },
      {
        "comment_id": "2032627919",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 7710,
        "pr_file": "components/admission-webhook/manifests/base/deployment.yaml",
        "discussion_id": "2031568051",
        "commented_code": "@@ -18,6 +18,15 @@ spec:\n         ports:\n         - name: https-webhook\n           containerPort: 4443\n+        securityContext:\n+          runAsNonRoot: true\n+          allowPrivilegeEscalation: false\n+          runAsUser: 1000",
        "comment_created_at": "2025-04-08T08:01:15+00:00",
        "comment_author": "juliusvonkohout",
        "comment_body": "So far 1000 works in our tests in https://github.com/kubeflow/manifests/pull/3050. I mean we can of course change it to 65532 if it works and you like that more, but having a proper securitycontext first is a massive improvement and then in a follow up PR we can do refinements.",
        "pr_file_module": null
      },
      {
        "comment_id": "2032772944",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 7710,
        "pr_file": "components/admission-webhook/manifests/base/deployment.yaml",
        "discussion_id": "2031568051",
        "commented_code": "@@ -18,6 +18,15 @@ spec:\n         ports:\n         - name: https-webhook\n           containerPort: 4443\n+        securityContext:\n+          runAsNonRoot: true\n+          allowPrivilegeEscalation: false\n+          runAsUser: 1000",
        "comment_created_at": "2025-04-08T09:21:29+00:00",
        "comment_author": "juliusvonkohout",
        "comment_body": "@akagami-harsh can you check whether 65532 works as well?",
        "pr_file_module": null
      },
      {
        "comment_id": "2032847785",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 7710,
        "pr_file": "components/admission-webhook/manifests/base/deployment.yaml",
        "discussion_id": "2031568051",
        "commented_code": "@@ -18,6 +18,15 @@ spec:\n         ports:\n         - name: https-webhook\n           containerPort: 4443\n+        securityContext:\n+          runAsNonRoot: true\n+          allowPrivilegeEscalation: false\n+          runAsUser: 1000",
        "comment_created_at": "2025-04-08T10:03:05+00:00",
        "comment_author": "akagami-harsh",
        "comment_body": "Yes it works, so changed user to 65532",
        "pr_file_module": null
      },
      {
        "comment_id": "2032891985",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 7710,
        "pr_file": "components/admission-webhook/manifests/base/deployment.yaml",
        "discussion_id": "2031568051",
        "commented_code": "@@ -18,6 +18,15 @@ spec:\n         ports:\n         - name: https-webhook\n           containerPort: 4443\n+        securityContext:\n+          runAsNonRoot: true\n+          allowPrivilegeEscalation: false\n+          runAsUser: 1000",
        "comment_created_at": "2025-04-08T10:28:41+00:00",
        "comment_author": "akagami-harsh",
        "comment_body": "https://github.com/kubeflow/manifests/actions/runs/14330485734/job/40165118566?pr=3050",
        "pr_file_module": null
      }
    ]
  },
  {
    "discussion_id": "798028619",
    "pr_number": 6241,
    "pr_file": "components/crud-web-apps/jupyter/manifests/base/cluster-role.yaml",
    "created_at": "2022-02-02T21:30:29+00:00",
    "commented_code": "metadata:\n  name: cluster-role\nrules:\n- apiGroups:\n  - \"\"\n  resources:\n  - namespaces\n  verbs:\n  - get\n  - list",
    "repo_full_name": "kubeflow/kubeflow",
    "discussion_comments": [
      {
        "comment_id": "798028619",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 6241,
        "pr_file": "components/crud-web-apps/jupyter/manifests/base/cluster-role.yaml",
        "discussion_id": "798028619",
        "commented_code": "@@ -3,15 +3,6 @@ kind: ClusterRole\n metadata:\n   name: cluster-role\n rules:\n-- apiGroups:\n-  - \"\"\n-  resources:\n-  - namespaces\n-  verbs:\n-  - get\n-  - list",
        "comment_created_at": "2022-02-02T21:30:29+00:00",
        "comment_author": "thesuperzapper",
        "comment_body": "@juliusvonkohout Can you confirm that the jupyter-web-app still functions correctly without \"get\" and \"list\" on namespaces?",
        "pr_file_module": null
      },
      {
        "comment_id": "799020047",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 6241,
        "pr_file": "components/crud-web-apps/jupyter/manifests/base/cluster-role.yaml",
        "discussion_id": "798028619",
        "commented_code": "@@ -3,15 +3,6 @@ kind: ClusterRole\n metadata:\n   name: cluster-role\n rules:\n-- apiGroups:\n-  - \"\"\n-  resources:\n-  - namespaces\n-  verbs:\n-  - get\n-  - list",
        "comment_created_at": "2022-02-03T22:25:50+00:00",
        "comment_author": "juliusvonkohout",
        "comment_body": "That seems to be the case. I just updated the roles, restarted jupyter-web-app and notebook-controller and was able to start, stop, create, delete jupyterlabs. There  are also no errors in the logs.\r\n\r\n[jupyter-web-app-deployment-7f7f55f9d8-slb8k-jupyter-web-app.log](https://github.com/kubeflow/kubeflow/files/7998674/jupyter-web-app-deployment-7f7f55f9d8-slb8k-jupyter-web-app.log)\r\n\r\n",
        "pr_file_module": null
      },
      {
        "comment_id": "799021186",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 6241,
        "pr_file": "components/crud-web-apps/jupyter/manifests/base/cluster-role.yaml",
        "discussion_id": "798028619",
        "commented_code": "@@ -3,15 +3,6 @@ kind: ClusterRole\n metadata:\n   name: cluster-role\n rules:\n-- apiGroups:\n-  - \"\"\n-  resources:\n-  - namespaces\n-  verbs:\n-  - get\n-  - list",
        "comment_created_at": "2022-02-03T22:27:42+00:00",
        "comment_author": "juliusvonkohout",
        "comment_body": "Is there more to test or it is ready to be merged?",
        "pr_file_module": null
      },
      {
        "comment_id": "799369886",
        "repo_full_name": "kubeflow/kubeflow",
        "pr_number": 6241,
        "pr_file": "components/crud-web-apps/jupyter/manifests/base/cluster-role.yaml",
        "discussion_id": "798028619",
        "commented_code": "@@ -3,15 +3,6 @@ kind: ClusterRole\n metadata:\n   name: cluster-role\n rules:\n-- apiGroups:\n-  - \"\"\n-  resources:\n-  - namespaces\n-  verbs:\n-  - get\n-  - list",
        "comment_created_at": "2022-02-04T11:05:40+00:00",
        "comment_author": "kimwnasptd",
        "comment_body": "Should be good to go. It would also be a bug from the current code if it couldn't operate without listing namespaces, since it shouldn't need these permissions in the first place.\r\n\r\nNote that we might need to re-introduce them though, in a future iteration when we would want the apps to be self-standing. But we need to discuss other items for this effort. But I'm also mentioning this context to keep in mind",
        "pr_file_module": null
      }
    ]
  }
]